src/bootstrap/bootstrap.py

```diff
@@ -954,6 +954,13 @@ def build_bootstrap_cmd(self, env):
         if deny_warnings:
             env["RUSTFLAGS"] += " -Dwarnings"
 
+        # Add RUSTFLAGS_BOOTSTRAP to RUSTFLAGS for bootstrap compilation.
+        # Note that RUSTFLAGS_BOOTSTRAP should always be added to the end of
+        # RUSTFLAGS to be actually effective (e.g., if we have `-Dwarnings` in
+        # RUSTFLAGS, passing `-Awarnings` from RUSTFLAGS_BOOTSTRAP should override it).
+        if "RUSTFLAGS_BOOTSTRAP" in env:
+            env["RUSTFLAGS"] += " " + env["RUSTFLAGS_BOOTSTRAP"]
```

Comment:
Won't this overwrite the RUSTFLAGS read by future Cargo invocations as part of the main build? That doesn't feel right to me if so.

Reply:
No, those only accessible from the stage0 cargo for bootstrapping the rust bootstrap.

You probably confused with this:

https://github.com/rust-lang/rust/blob/89d610cd06f90cd6bd0bccdb395a3127290e1b45/src/bootstrap/bootstrap.py#L1107-L1110

Reply:
Ah, I see. Okay, sounds good.

Reply:
Non of the environment variables defined in that(`def build_bootstrap_cmd`) function can be read from rust-side bootstrapping.